### PR TITLE
`transaction` detail

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
see detail and link to explorer

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
image

### CODE REVIEW

File: screens/AppNavigator/screens/TransactionsScreen/screens/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  desc: string // of each transaction type, eg: Sent, Add Liquidity
+  iconName: string
+  color: string
+  amount: string
+  block: number
+  token: string
+  txid: string
+}
+
+export function activitiesToViewModel (activities: AddressActivity[], navigation: NavigationProp<ParamListBase>): VMTransaction[] {
```
Comment (ivan-zynesis) [inline screens/AppNavigator/screens/TransactionsScreen/screens/reducer.ts:17]: Didn't actually done any automated test for FE code, I should start here?
(these functions actually only used by the transaction listing page, only the view model is shared)

@fuxingloh @thedoublejay

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionsScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import { useEffect, useState } from 'react'
+import { useWalletAPI } from '../../../../../hooks/wallet/WalletAPI'
+import { useWhaleApiClient } from '../../../../../hooks/api/useWhaleApiClient'
+import { Ionicons } from '@expo/vector-icons'
+import { Text, NumberText } from '../../../../../components'
+import { useNavigation } from '@react-navigation/native'
+import { activitiesToViewModel, VMTransaction } from './reducer'
+
+interface TxRow extends VMTransaction {
+  onPress: () => void
```
Comment (ivan-zynesis) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionsScreen.tsx:14]: Split this out, VMTransaction also the prop being passed to detail page just without the row press function.

File: screens/AppNavigator/screens/TransactionsScreen/TransactionsNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      <TransactionsStack.Screen
+        name='Transactions'
+        component={TransactionsScreen}
+        options={{ headerTitle: translate('screens/TransactionsScreen', 'Transactions') }}
+      />
+      <TransactionsStack.Screen
+        name='TransactionDetail'
+        component={TransactionDetailScreen}
+        options={{
+          headerTitle: translate('screens/TransactionsDetailScreen', 'TransactionDetail')
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/TransactionsNavigator.tsx:33]:
```suggestion
          headerTitle: translate('screens/TransactionsDetailScreen', 'Transaction')
```

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          <View style={tailwind('w-1/2 flex-1')}>
+            <Text style={tailwind('font-medium text-right text-gray-600')}>{rhs}</Text>
+          </View>
+        </View>
+      </View>
+    )
+  }
+
+  const url = explorerUrl(tx.txid)
+  const onTxidUrlPressed = React.useCallback(async () => {
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:36]: I think it's better to import the hook directly

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    if (supported) {
+      await Linking.openURL(url)
+    }
+  }, [url])
+
+  return (
+    <View>
+      {RenderRow('Type', translate('screens/TransactionDetailScreen', tx.desc))}
+      {/* TODO(@ivan-zynesis): handle different transaction type other than sent/receive */}
+      {RenderRow('Amount', translate('screens/TransactionDetailScreen', tx.amount))}
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:47]: Missing unit here

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      {grayDivider}
+      <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('flex-1 flex-row flex-initial')}>
+          <View style={tailwind('flex-1')}>
+            <Text style={tailwind('flex-1 font-medium text-gray-500')}>
+              {translate('screens/TransactionDetailScreen', tx.txid)}
+            </Text>
+          </View>
+          <View style={tailwind('w-8 flex-grow-0 justify-center')}>
+            <Ionicons name='open-outline' size={24} color='pink' onPress={onTxidUrlPressed} />
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:58]: Color should be #FF00AF

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+export function TransactionDetailScreen (props: Props): JSX.Element {
+  const { tx } = props.route.params
+
+  const grayDivider = <View style={tailwind('bg-gray w-full h-4')} />
+  const RenderRow = (lhs: string, rhs: string): JSX.Element => {
+    return (
+      <View>
+        {grayDivider}
+        <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:23]: Add more padding to rows, it's too close to the edges

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      {RenderRow('Block', translate('screens/TransactionDetailScreen', `${tx.block}`))}
+      {grayDivider}
+      <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('flex-1 flex-row flex-initial')}>
+          <View style={tailwind('flex-1')}>
+            <Text style={tailwind('flex-1 font-medium text-gray-500')}>
+              {translate('screens/TransactionDetailScreen', tx.txid)}
+            </Text>
+          </View>
+          <View style={tailwind('w-8 flex-grow-0 justify-center')}>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:57]: add some margin between the txid and url button.

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+import tailwind from 'tailwind-rn'
+import { translate } from '../../../../../translations'
+import { View, Linking } from 'react-native'
+import { Text } from '../../../../../components'
+import { StackScreenProps } from '@react-navigation/stack'
+import { TransactionsParamList } from '../TransactionsNavigator'
+import { Ionicons } from '@expo/vector-icons'
+
+// FIXME(@ivan-zynesis): envvar
+const EXPLORER_BASE_URL = 'https://explorer.defichain.io'
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:11]: Should this one be updated based on network?

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx (hunk 9 above)
Comment (ivan-zynesis) [reply screens/AppNavigator/screens/TransactionsScreen/screens/TransactionDetailScreen.tsx:11]: Yeah, thus the FIXME is there. As we are working on new explorer, so the link here isnt really finalized (for every env).

File: cypress/integration/functional/transaction/list.spec.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    before(() => {
+      cy.createEmptyWallet()
+
+      // setup transactions history, topup twice
+      const baseURL = '/v1/playground/rpc'
+      cy.intercept(`${baseURL}/sendtoaddress`).as('sendToAddress')
+      cy.getByTestID('playground_wallet_top_up').click()
+      cy.wait(['@sendToAddress'])
+      cy.getByTestID('playground_wallet_top_up').click()
+      cy.wait(['@sendToAddress'])
```
Comment (fuxingloh) [inline cypress/integration/functional/transaction/list.spec.ts:12]: Look at the most recent code, this part has changed.

**Review by thedoublejay — DISMISSED** [review]

**Comment by fuxingloh** [comment]
ps there is conflicts again

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**